lib/src/config/config.rs

```diff
@@ -379,7 +379,7 @@ impl Config {
     /// ```
     pub fn set_address<A: Into<String>>(&mut self, address: A) -> Result<()> {
         let address = address.into();
-        if (&*address, 0u16).to_socket_addrs().is_err() {
+        if !address.starts_with("unix://") && (&*address, 0u16).to_socket_addrs().is_err() {
```

Comment:
We also need to validate the path here and check if this is being done on Windows. All configuration validation needs to happen at this point. My guess is that we really want to store the address as an `enum` with variants for TCP, Unix sockets, and even UDP, if it's trivial to add. Something like:

```rust
enum Address {
    Tcp(SocketAddr),
    Udp(SocketAddr),
    /// Stores the validated path to the unix domain socket.
    Unix(PathBuf)
}
```

If there's a better type to use than `PathBuf`, even better.